conda_forge_tick/migrators.xsh

```diff
@@ -1169,3 +1169,105 @@ class GFortranOSXRebuild(Rebuild):
 
     def pr_title(self):
         return "Rebuild for gfortran 7 for OSX"
+
+
+# This may replace Rebuild
+class MigrationYaml(Migrator):
+    """Migrator for bumping the build number."""
+    migrator_version = 0
+    rerender = True
+    bump_number = 1
+    # TODO: add a description kwarg for the status page at some point.
+    # TODO: make yaml_contents an arg?
+    def __init__(self, yaml_contents='',
+                 graph=None, name=None, pr_limit=0, top_level=None,
+                 cycles=None, obj_version=None):
+        super().__init__(pr_limit, obj_version)
+        self.yaml_contents = yaml_contents
+        if graph == None:
+            self.graph = nx.DiGraph()
+        else:
+            self.graph = graph
+        self.name = name
+        self.top_level = top_level
+        self.cycles = set(chain.from_iterable(cycles or []))
+
+    def filter(self, attrs):
+        if super().filter(attrs, 'Upstream:'):
+            return True
+        if attrs['feedstock_name'] not in self.graph:
+            return True
+        # If in top level or in a cycle don't check for upstreams just build
+        if ((self.top_level and attrs['feedstock_name'] in self.top_level)
+            or (self.cycles and attrs['feedstock_name'] in self.cycles)):
+            return False
+        # Check if all upstreams have been built
+        for node in self.graph.predecessors(attrs['feedstock_name']):
+            att = self.graph.node[node]
+            muid = frozen_to_json_friendly(self.migrator_uid(att))
+            if muid not in att.get('PRed', []) and not att.get('archived', False):
+                return True
+            # This is due to some PRed_json loss due to bad graph deploy outage
+            for m_pred_json in att.get('PRed_json', []):
+                if m_pred_json['data'] == muid['data']:
+                    break
+            else:
+                m_pred_json = None
+            if m_pred_json and m_pred_json['PR'].get('state', '') == 'open':
+                return True
+        return False
+
+    def migrate(self, recipe_dir, attrs, **kwargs):
+        with indir(os.path.join(recipe_dir, '..')):
+            os.makedirs('migrations', exist_ok=True)
```

Comment:
Can we make this hidden like all th other folders except for recipe folder?

Reply:
This is inside the recipe folder

Reply:
Actually I take that back. This is putting the folder where conda-smithy expects it so it needs to be called what conda-smithy expects.

Reply:
I raised this in conda-smithy PR, but I was ignored there. That's why I'm raising this here